In [1]:
# Mengimport Modul dan Pustaka yang akan dipakai untuk melakukan penelitian
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np

# Membaca file CSV 'airplane_price_dataset.csv' dan memuatnya ke dalam DataFrame 'df'
df = pd.read_csv('airplane_price_dataset.csv')
df

Model  Üretim Yılı  Motor Sayısı Motor Türü  Kapasite  \
0      Bombardier CRJ200         1987             2   Turbofan        50   
1      Bombardier CRJ200         1997             2   Turbofan        50   
2            Airbus A320         1988             2   Turbofan       180   
3             Boeing 737         2023             2   Turbofan       162   
4             Cessna 172         1985             1     Piston         4   
...                  ...          ...           ...        ...       ...   
12372         Boeing 777         1994             2   Turbofan       396   
12373  Bombardier CRJ200         2005             2   Turbofan        50   
12374         Boeing 737         2015             2   Turbofan       162   
12375         Cessna 172         2013             1     Piston         4   
12376  Bombardier CRJ200         2013             2   Turbofan        50   

       Menzil (km)  Yakıt Tüketimi (L/saat)  Saatlik Bakım Maliyeti ($)  Yaş  \
0             3000                    14.36                     2185.43   36   
1             3000                     4.03                     1202.08   26   
2             6300                    13.26                      761.38   35   
3             5700                    14.61                      592.63    0   
4             1285                    18.49                     4245.99   38   
...            ...                      ...                         ...  ...   
12372        15600                    12.16                     3054.29   29   
12373         3000                     8.26                     3107.67   18   
12374         5700                     6.93                     3972.16    8   
12375         1285                    14.28                      999.84   10   
12376         3000                     2.42                     1035.75   10   

       Satış Bölgesi     Fiyat ($)  
0               Asya  1.285708e+07  
1             Avrupa  1.391406e+07  
2         Avustralya  9.073570e+07  
3         Avustralya  1.366597e+08  
4      Güney Amerika  2.037981e+05  
...              ...           ...  
12372         Avrupa  3.898071e+08  
12373         Afrika  1.593888e+07  
12374  Kuzey Amerika  9.870994e+07  
12375         Afrika  3.965207e+05  
12376     Avustralya  2.320782e+07  

[12377 rows x 11 columns]

In [2]:
print(df.columns)

Index(['Model', 'Üretim Yılı', 'Motor Sayısı', 'Motor Türü', 'Kapasite',
       'Menzil (km)', 'Yakıt Tüketimi (L/saat)', 'Saatlik Bakım Maliyeti ($)',
       'Yaş', 'Satış Bölgesi', 'Fiyat ($)'],
      dtype='object')


In [3]:
# Translate columns to English
translation_columns = {
"Model": "Model",
"Üretim Yılı": "Year of Production",
"Motor Sayısı": "Number of Engines",
"Motor Türü": "Engine Type",
"Kapasite": "Capacity",
"Menzil (km)": "Range in km",
"Yakıt Tüketimi (L/saat)": "Fuel consumption in (L/hour)",
"Saatlik Bakım Maliyeti ($)": "Maintenance cost per hour",
"Yaş": "Age",
"Satış Bölgesi": "Sales Region",
"Fiyat ($)": "Price in USD"
}

# Rename the columns
df_airplane = df.rename(columns=translation_columns)

# Format the Price column in USD for better visualization
pd.options.display.float_format = "{:,.2f}".format

display(df_airplane)

Model  Year of Production  Number of Engines Engine Type  \
0      Bombardier CRJ200                1987                  2    Turbofan   
1      Bombardier CRJ200                1997                  2    Turbofan   
2            Airbus A320                1988                  2    Turbofan   
3             Boeing 737                2023                  2    Turbofan   
4             Cessna 172                1985                  1      Piston   
...                  ...                 ...                ...         ...   
12372         Boeing 777                1994                  2    Turbofan   
12373  Bombardier CRJ200                2005                  2    Turbofan   
12374         Boeing 737                2015                  2    Turbofan   
12375         Cessna 172                2013                  1      Piston   
12376  Bombardier CRJ200                2013                  2    Turbofan   

       Capacity  Range in km  Fuel consumption in (L/hour)  \
0            50         3000                         14.36   
1            50         3000                          4.03   
2           180         6300                         13.26   
3           162         5700                         14.61   
4             4         1285                         18.49   
...         ...          ...                           ...   
12372       396        15600                         12.16   
12373        50         3000                          8.26   
12374       162         5700                          6.93   
12375         4         1285                         14.28   
12376        50         3000                          2.42   

       Maintenance cost per hour  Age   Sales Region   Price in USD  
0                       2,185.43   36           Asya  12,857,083.91  
1                       1,202.08   26         Avrupa  13,914,058.55  
2                         761.38   35     Avustralya  90,735,695.86  
3                         592.63    0     Avustralya 136,659,689.37  
4                       4,245.99   38  Güney Amerika     203,798.10  
...                          ...  ...            ...            ...  
12372                   3,054.29   29         Avrupa 389,807,070.36  
12373                   3,107.67   18         Afrika  15,938,876.07  
12374                   3,972.16    8  Kuzey Amerika  98,709,939.20  
12375                     999.84   10         Afrika     396,520.67  
12376                   1,035.75   10     Avustralya  23,207,818.82  

[12377 rows x 11 columns]

In [5]:
print(df_airplane)

print(df_airplane.columns)

                   Model  Year of Production  Number of Engines Engine Type  \
0      Bombardier CRJ200                1987                  2    Turbofan   
1      Bombardier CRJ200                1997                  2    Turbofan   
2            Airbus A320                1988                  2    Turbofan   
3             Boeing 737                2023                  2    Turbofan   
4             Cessna 172                1985                  1      Piston   
...                  ...                 ...                ...         ...   
12372         Boeing 777                1994                  2    Turbofan   
12373  Bombardier CRJ200                2005                  2    Turbofan   
12374         Boeing 737                2015                  2    Turbofan   
12375         Cessna 172                2013                  1      Piston   
12376  Bombardier CRJ200                2013                  2    Turbofan   

       Capacity  Range in km  Fuel consumption in (

**Nama_kolom**

    1. Model Pesawat (Model): Model pesawat, seperti Boeing 737 atau Airbus A320, yang mempengaruhi harga berdasarkan merek dan kemampuan kinerja.

    2. Tahun Produksi (Production Year): Tahun pembuatan pesawat, yang membantu dalam perhitungan usia pesawat dan depresiasi harga.

    3. Jumlah Mesin (Number of Engines): Jumlah mesin pada pesawat, bisa 1 untuk mesin piston, 2 atau lebih untuk mesin lainnya.

    4. Jenis Mesin (Engine Type): Tipe mesin, seperti Turbofan atau Piston, yang mempengaruhi kinerja, efisiensi bahan bakar, dan harga.

    5. Kapasitas Penumpang (Capacity): Kapasitas penumpang pesawat. Pesawat dengan kapasitas lebih besar umumnya memiliki harga dasar yang lebih tinggi.

    6. Jangkauan (km) (Range in km): Jarak maksimum yang dapat ditempuh pesawat tanpa perlu pengisian bahan bakar ulang. Jangkauan yang lebih jauh meningkatkan fleksibilitas operasional dan biaya.

    7. Konsumsi Bahan Bakar (L/jam) (Fuel Consumption in L/hour): Rata-rata konsumsi bahan bakar pesawat per jam. Mesin Turbofan lebih efisien dalam konsumsi bahan bakar dibandingkan mesin piston.

    8. Biaya Pemeliharaan per Jam ($) (Hourly Maintenance Cost): Biaya pemeliharaan rata-rata per jam dalam USD. Biaya pemeliharaan yang tinggi dapat menurunkan permintaan dan mempengaruhi harga.

    9. Usia Pesawat (Age): Usia pesawat, dihitung dengan rumus 2023 - Tahun Produksi. Pesawat yang lebih tua cenderung memiliki harga yang lebih rendah karena depresiasi.

    10. Wilayah Penjualan (Sales Region): Wilayah tempat pesawat dijual (misalnya Asia, Eropa, Amerika Utara). Permintaan regional dan variasi mata uang dapat mempengaruhi harga.

    11. Harga ($) (Price in USD): Harga akhir pesawat dalam USD, yang menjadi variabel target untuk prediksi.
    

In [6]:
# Visualizing dataframe information with info
df_airplane.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12377 entries, 0 to 12376
Data columns (total 11 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Model                         12377 non-null  object 
 1   Year of Production            12377 non-null  int64  
 2   Number of Engines             12377 non-null  int64  
 3   Engine Type                   12377 non-null  object 
 4   Capacity                      12377 non-null  int64  
 5   Range in km                   12377 non-null  int64  
 6   Fuel consumption in (L/hour)  12377 non-null  float64
 7   Maintenance cost per hour     12377 non-null  float64
 8   Age                           12377 non-null  int64  
 9   Sales Region                  12377 non-null  object 
 10  Price in USD                  12377 non-null  float64
dtypes: float64(3), int64(5), object(3)
memory usage: 1.0+ MB


In [7]:
df_airplane.isna().sum()

Model                           0
Year of Production              0
Number of Engines               0
Engine Type                     0
Capacity                        0
Range in km                     0
Fuel consumption in (L/hour)    0
Maintenance cost per hour       0
Age                             0
Sales Region                    0
Price in USD                    0
dtype: int64

In [8]:
# Check if there are duplicate values ​​in the dataframe
df_airplane.duplicated().sum()

np.int64(0)